src/main/scala/s_mach/similar/SimilarOps.scala

```diff
@@ -1,16 +1,65 @@
 package s_mach.similar
 
 object SimilarOps {
+
+  def simByDistanceThreshold[A](maxDistance : Int)(dist : (A,A) => Int): CanSimilar[A] = {
+    new CanSimilar[A] {
+      override def similar(a1: A, a2: A): Double = {
+        dist(a1, a2) match {
+          case 0 => 1.0
+          case within if within <= maxDistance => (maxDistance - within)*1.0 / maxDistance
+          case _ => 0
+        }
+      }
+    }
+  }
+
+  // this needs to calculate the union size somehow...
   // http://en.wikipedia.org/wiki/Jaccard_index
-  def calcJaccardIndex(a_intersect_b_size: Int, a_size: Int, b_size: Int) : Double = ???
+  def calcJaccardIndex(a_intersect_b_size: Int, a_size: Int, b_size: Int) : Double = {
+    if (a_size == 0 || b_size == 0) 1 else {
+      a_intersect_b_size / (a_size + b_size)
+    }
```

Comment:
Not sure but you may need a .toDouble on either a_size or b_size to prevent Int divded by Int operation here (Int divide by Int rounds to whole number)